File: Algorithms/Greedy_algorithms/activity_selection.py
# Given a list of activities with their start and finish times, select the maximum of
# non-overlapping activities that can be performed

def print_max_activities(start, finish):
    n = len(start)
    print("Selected activities:", end=" ")
    i = 0
    print(i, end=" ")

    for j in range(1, n):
        if start[j] >= finish[i]:
            print(j, end=" ")
            i = j

start = [1, 3, 0, 5, 8, 5]
finish = [2, 4, 6, 7, 9, 9]
print_max_activities(start, finish)
print("\n")

File: Algorithms/Greedy_algorithms/activity_selection.c
// Given a list of activities with their start and finish times, select the maximum of
// non-overlapping activities that can be performed

#include <stdio.h>

void printMaxActivities(int start[], int finish[], int n) {
    int i, j;
    printf("Selected activities: ");
    i = 0;
    printf("%d ", i);

    for (j = 1; j < n; j++) {
        if (start[j] >= finish[i]) {
            printf("%d ", j);
            i = j;
        }
    }
}

int main() {
    int start[] = {1, 3, 0, 5, 8, 5};
    int finish[] = {2, 4, 6, 7, 9, 9};
    int n = sizeof(start) / sizeof(start[0]);
    printMaxActivities(start, finish, n);
    printf("\n");
    return 0;
}
